### DISCUSSION TITLE
A disabled query returns stale data?

### DISCUSSION THREAD

**Discussion by JanStevens** [TanStack/query#2180] [Q&A]
Hi 👋 ,

I have a component that allows to search for `Contacts` and / or `ContactGroups` depending on two booleans: `withContacts` and `withContactGroups`. The following code is used to fetch `contacts` and `contactGroups`:

```ts
  const { data: contacts, isFetching: contactsFetching } = useContacts(
    { search: debouncedSearchValue },
    { enabled: withContacts, keepPreviousData: true },
  );

  const { data: contactGroups, isFetching: groupsFetching } = useContactGroups(
    { search: debouncedSearchValue },
    { enabled: withContactGroups, keepPreviousData: true },
  );
```

`useContacts` and `useContactGroups` are a simple custom hook wrapper around `useQuery` example:

```ts
export const useContacts = (
  params: ContactSearchParams,
  config?: UseQueryOptions<ApiResponse<Contact[]>>,
) => {
  const user = useSelector(getUser);

  const { data, status, isLoading, isFetching } = useQuery(
    ['contacts', params],
    () => fetchContacts(user.id, params),
    config
  );

  return {
    data: data?.data || [],
    status,
    isLoading,
    isFetching,
    total: data?.meta?.total || 0,
  };
};
```

What I notice is that if I first open my component for showing groups (`withContacts: true, withContactGroups: true`) I see indeed only groups query being executed. 
I then navigate to another page and open the same component but with `withContacts: true, withContactGroups: false` I see that contacts query get executed but **contactGroups query also returns it stale data even when its disabled**

Now my question is, is this expected behaviour or not, I'm I missing something obvious here?

I made a code sandbox to demo this behaviour: https://codesandbox.io/s/react-query-enabled-queries-3tiz9?file=/src/index.js
I would expect that if the query is **disabled** the return data is also `undefined`, but it returns the stale data instead.

**Comment by TkDodo** [comment]
if data is available in the cache, react-query will give it to you, no matter what.

since you are in control of the enabled flag, you can always chose to not display that data if your query is disabled :)

**Reply by JanStevens** [reply]
That surprised me as I would expect react query not to return stale data.
